src/main/java/com/amazonaws/kinesisvideo/internal/producer/StreamEventMetadata.java

```diff
@@ -0,0 +1,60 @@
+package com.amazonaws.kinesisvideo.internal.producer;
+
+import javax.annotation.Nullable;
+
+/**
+ * Kinesis Video stream event metadata as provided by the native codebase.
+ */
+public class StreamEventMetadata {
+
+    // Current class version as described by the native struct.
+    public static final int STREAM_EVENT_METADATA_CURRENT_VERSION = 0;
+
+
+    // Version of the struct.
+    int mVersion;
+
+    // Optional s3 prefix.
+    @Nullable
+    String mImagePrefix;
+
+    // Optional optimization stating how many name/value pairs to be appended.
+    // NOTE: This is NOT optional if using mNames and mValues, else the arrays will not be used.
+    @Nullable
+    byte mNumberOfPairs;
```

Comment:
Isn't int more suitable for numberOfPairs? Its max value is 10

Reply:
Max byte value is 127, it is a byte in C, so keeping it the same.

Reply:
Do we do this everywhere we map a C Struct with a UINT8 that we convert it to a byte?  Is there some reason the number of bytes for each parameter needs to match?  If not.. I think it should be an int as well something called `numberOfPairs` just looks very odd if it's a byte.

Reply:
There actually isn't any good existing example of a byte member from a pic struct being used in Java - all of the other places we use byte members are only for pic-internal structs, so no precedent to work off of.

I don't have a strong opinion on either way, so I'll go ahead and switch it to an int with the benefit  being that it will be more clear that this member is a numeric value.